src/vivarium_gates_mncnh/components/intervention.py

```diff
@@ -1,74 +1,93 @@
 from __future__ import annotations
 
-from functools import partial
-
 import pandas as pd
 from vivarium import Component
 from vivarium.framework.engine import Builder
 
 from vivarium_gates_mncnh.constants import data_values
-from vivarium_gates_mncnh.constants.data_keys import NO_CPAP_RISK
 from vivarium_gates_mncnh.constants.data_values import COLUMNS, PIPELINES
-from vivarium_gates_mncnh.utilities import get_location
 
 
-class NoCPAPRisk(Component):
-    """Component for CPAP intervention. This is essentially a risk effect."""
+class NeonatalNoInterventionRisk(Component):
+    """Component that modifies a neonatal CSMR pipeline based on the lack of an intervention.
+    This is the implementation of the RiskEffect for these dichoctomous risks."""
+
+    INTERVENTION_PIPELINE_MODIFIERS_MAP = {
+        "cpap": PIPELINES.PRETERM_WITH_RDS_FINAL_CSMR,
+        "antibiotics": PIPELINES.NEONATAL_SEPSIS_FINAL_CSMR,
+    }
 
     @property
     def configuration_defaults(self) -> dict:
         return {
             self.name: {
                 "data_sources": {
-                    "relative_risk": NO_CPAP_RISK.RELATIVE_RISK,
+                    "relative_risk": self.load_relative_risk_data,
                     "paf": self.load_paf_data,
                 }
             }
         }
 
     @property
     def columns_required(self) -> list[str]:
-        return [COLUMNS.CPAP_AVAILABLE]
+        return [self.col_required]
 
-    def __init__(self) -> None:
+    def __init__(
+        self,
+        lack_of_intervention_risk: str,
+    ) -> None:
         super().__init__()
-        self.preterm_csmr_target = PIPELINES.PRETERM_WITH_RDS_FINAL_CSMR
+        self.lack_of_intervention_risk = lack_of_intervention_risk
+        self.col_required = f"{lack_of_intervention_risk}_available"
 
     def setup(self, builder: Builder) -> None:
         self.randomness = builder.randomness.get_stream(self.name)
-        self.location = get_location(builder)
-        self.preterm_with_rds_csmr = builder.value.get_value(self.preterm_csmr_target)
+        self.csmr_target_pipeline_name = self.get_target_pipeline_name()
         builder.value.register_value_modifier(
-            self.preterm_with_rds_csmr.name,
-            self.modify_preterm_with_rds_csmr,
+            self.csmr_target_pipeline_name,
+            self.modify_csmr_pipeline,
+            component=self,
             required_resources=[
                 COLUMNS.DELIVERY_FACILITY_TYPE,
-                COLUMNS.CPAP_AVAILABLE,
+                self.col_required,
```

Comment:
You should also be requiring the columns that the lookup table needs.

Random note: `LookupTables` should probably be resources that need to be specified as required resources just like pipelines and columns.